sdk/keyvault/azure-keyvault-keys/tests/test_crypto_client.py

```diff
@@ -491,6 +491,82 @@ def test_calls_service_for_operations_unsupported_locally():
     assert supports_nothing.wrap_key.call_count == 0
 
 
+def test_local_only_mode_no_service_calls():
+    """A local-only CryptographyClient shouldn't call the service if an operation can't be performed locally"""
+
+    mock_client = mock.Mock()
+    jwk = mock.Mock(spec=JsonWebKey)
+    client = CryptographyClient.from_jwk(jwk=jwk)
```

Comment:
We expect no request, so I would use a mock transport that raises when the client tries to send e.g.
```py
mock.Mock(send=mock.Mock(side_effect=Exception("client shouldn't send a request"))
```

Reply:
(That is, if it would be a bug for the client to send a request in local-only mode.)

Reply:
This is one that I've made a note of to revisit -- I'm not too familiar with Mock and wasn't able to randomly turn enough dials to get a better understanding. I figure that asserting that the underlying client doesn't call get_key is sufficient for the time being